superset/config.py

```diff
@@ -576,31 +578,44 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
 # Setup image size default is (300, 200, True)
 # IMG_SIZE = (300, 200, True)
 
-# Default cache timeout, applies to all cache backends unless specifically overridden in
+
+# Default cache config, applies to all cache backends unless specifically overridden in
 # each cache config.
-CACHE_DEFAULT_TIMEOUT = int(timedelta(days=1).total_seconds())
+def DEFAULT_CACHE_CONFIG_FUNC(  # pylint: disable=invalid-name
+    app: Flask,
+) -> CacheConfig:
+    default_timeout = app.config.get("CACHE_DEFAULT_TIMEOUT")
+    if default_timeout is None:
+        default_timeout = int(timedelta(days=1).total_seconds())
+    else:
+        logger.warning(
+            "The global config flag CACHE_DEFAULT_TIMEOUT has been deprecated "
+            "and will be removed in Superset 2.0. Please set default cache options in "
+            "DEFAULT_CACHE_CONFIG_FUNC"
+        )
+
+    return {
+        "CACHE_TYPE": "SimpleCache" if app.debug else "NullCache",
+        "CACHE_THRESHOLD": math.inf,
```

Comment:
It turns out `FileSystemCache` treats `"CACHE_THRESHOLD": 0` as "no threshold", while `SimpleCache` treats it as "expire everything".
- `SimpleCache`: https://github.com/pallets-eco/flask-caching/blob/058629ca4d38d1215fd911d41443751d0806bd4c/src/flask_caching/backends/simplecache.py#L61-L69
- `FileSystemCache`: https://github.com/pallets-eco/flask-caching/blob/058629ca4d38d1215fd911d41443751d0806bd4c/src/flask_caching/backends/filesystemcache.py#L128-L147

Both default to 500 if unset (The setting has no effect on other backends). To avoid a situation where someone wonders why their cache starts expiring entries before the timeout I decided to set `CACHE_THRESHOLD` to `math.inf`.

Reply:
To keep things simple, the default could be just a `NullCache` that does not work with the most appropriate values set for other cache options.

A `DEFAULT_CACHE_CONFIG_FUNC` factory is also confusing since ideally all of Superset should just use the same cache, we introduced so many different cache storages because they SHOULD  be configured differently in accord with their different purposes. If the all are using the same DEFAULT_CACHE, then it should just use the default cache storage.

Reply:
I removed the `DEFAULT_CACHE_CONFIG`